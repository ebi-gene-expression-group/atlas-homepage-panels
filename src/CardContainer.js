import React from 'react'
import PropTypes from 'prop-types'
import Card from './Card'
import LatestExperimentCardContainer from './LatestExperimentCardContainer'

const CardContainer = ({panelName, host, resource}) => {
  switch(panelName) {
    case `Featured`:
      return <Card host={host} resource={resource[`Featured`]}/>
      break
    case `Latest experiments`:
      return <LatestExperimentCardContainer host={host} resource={resource[`Latest`]}/>
      break
    default:
      return null
  } 
}

CardContainer.propTypes = {
  panelName: PropTypes.string.isRequired,
  host: PropTypes.string.isRequired,
  resource: PropTypes.object.isRequired
}

export default CardContainer
