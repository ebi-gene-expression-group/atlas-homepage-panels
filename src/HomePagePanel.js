import React from 'react'
import PropTypes from 'prop-types'
import Card from './Card'

class HomePagePanel extends React.Component {
  constructor(props) {
    super(props)
  }

  render() {
  	return (
  		<span>I am a list of cards </span>
  	)

  }
}

HomePagePanel.propTypes = {
  atlasUrl: PropTypes.string
}

export default HomePagePanel