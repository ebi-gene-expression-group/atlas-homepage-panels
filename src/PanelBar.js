import React from 'react'
import PropTypes from 'prop-types'

class PanelBar extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      panelName: this.props.panelNames[0]
    }

    this.changePanel = this.changePanel.bind(this)
  }

  changePanel(event){
    this.setState({
      panelName: event.id
    })
    this.props.onClick(event)
  }

  render() {
    var cardNames = this.props.panelNames

    var panelNameList = cardNames.map(cardName => 
      cardName==`Species` ?
        <li className="tabs-title" key={cardName}>
          {cardName==this.state.panelName ? 
          <a id={cardName.toLowerCase()} href="#by-species" aria-selected="true" onClick={this.changePanel}>By species</a> :
          <a id={cardName.toLowerCase()} href="#by-species" aria-selected="false" onClick={this.changePanel}>By species</a>}

        </li> :
        <li className="tabs-title" key={cardName}>
          {cardName==this.state.panelName ? 
            <a id={cardName.toLowerCase()} href="#by-animals" aria-selected="true" onClick={this.changePanel}>{cardName}</a> :
            <a id={cardName.toLowerCase()} href="#by-animals" aria-selected="false" onClick={this.changePanel}>{cardName}</a>}
        </li>
    )

    return (
      <div>
        <ul className="tabs" data-tabs id="browse-by-tabs">
          {panelNameList}
        </ul>
      </div>
    )
  }
}

PanelBar.propTypes = {
  atlasUrl: PropTypes.string
}

export default PanelBar
