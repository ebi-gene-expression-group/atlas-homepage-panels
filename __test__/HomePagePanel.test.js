import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme from 'enzyme'
import {shallow, mount, render} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import HomePagePanel from '../src/HomePagePanel'
import PanelBar from '../src/PanelBar'
import CardContainer from '../src/CardContainer'

Enzyme.configure({ adapter: new Adapter() })

describe(`HomePagePanel`, () => {
  const props = {panelNameAList:[`foo1`, `foo2`, `foo3`], panelNameBList: [`test1`, `test2`], host: `link` }
  test(`should have two panel boxes`, () => {
     expect(shallow(<HomePagePanel {...props}/>).find(PanelBar).length).toEqual(2)
     expect(shallow(<HomePagePanel {...props}/>).find(CardContainer).length).toEqual(2)
  })

  test(`renders different panel page after clicking each panel bar`, () => {
    const wrapper = shallow(<HomePagePanel {...props}/>)
    wrapper.find(PanelBar).at(0).simulate('click', { target: { id: 'UsernameA' } })
    wrapper.update()
    expect(wrapper.state().panelNameA).toEqual(`UsernameA`)

    wrapper.find(PanelBar).at(1).simulate('click', { target: { id: 'UsernameB' } })
    wrapper.update()
    expect(wrapper.state().panelNameB).toEqual(`UsernameB`)
  })

  test(`matches snapshot`, () => {
    const tree = renderer.create(<HomePagePanel {...props}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
