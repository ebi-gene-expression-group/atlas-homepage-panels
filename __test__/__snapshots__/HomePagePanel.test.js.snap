// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePagePanel matches snapshot 1`] = `
<div
  className="row expanded margin-top-large"
  data-equalizer={true}
>
  <div
    className="small-12 medium-12 large-6 columns"
  >
    <div
      className="callout experiment-list-latest padding-bottom-for-button"
      data-equalizer-watch={true}
    >
      <div>
        <ul
          className="tabs"
          data-tabs={true}
          id="browse-by-tabs"
        >
          <li
            className="tabs-title"
          >
            <a
              aria-selected="true"
              id="foo1"
              onClick={[Function]}
            >
              foo1
            </a>
          </li>
          <li
            className="tabs-title"
          >
            <a
              aria-selected="false"
              id="foo2"
              onClick={[Function]}
            >
              foo2
            </a>
          </li>
          <li
            className="tabs-title"
          >
            <a
              aria-selected="false"
              id="foo3"
              onClick={[Function]}
            >
              foo3
            </a>
          </li>
        </ul>
      </div>
      <div
        className="tabs-content"
        data-tabs-content="browse-by-tabs"
      >
        <div
          className="row margin-bottom-xxlarge"
        >
          <div
            className="row column"
          >
            <div
              className="callout alert small"
            >
              <h5>
                Oops!
              </h5>
              <p>
                There was a problem communicating with the server. Please try again later.
                <br />
                If the error persists, in order to help us debug the issue, please copy the URL and this message and send it to us via 
                <a
                  href="https://www.ebi.ac.uk/support/gxasc"
                >
                  the EBI Support & Feedback system
                </a>
                :
              </p>
              <code>
                ReferenceError: fetch is not defined
              </code>
            </div>
          </div>
        </div>
      </div>
      <div
        className="row align-row-to-bottom"
      >
        <div
          className="small-6 small-centered columns margin-top-large"
        >
          <a
            className="button float-center"
            href="linkexperiments"
            id="viewAllLink"
          >
            View all species
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="small-12 medium-12 large-6 columns"
  >
    <div
      className="callout experiment-list-latest padding-bottom-for-button"
      data-equalizer-watch={true}
    >
      <div>
        <ul
          className="tabs"
          data-tabs={true}
          id="browse-by-tabs"
        >
          <li
            className="tabs-title"
          >
            <a
              aria-selected="true"
              id="test1"
              onClick={[Function]}
            >
              test1
            </a>
          </li>
          <li
            className="tabs-title"
          >
            <a
              aria-selected="false"
              id="test2"
              onClick={[Function]}
            >
              test2
            </a>
          </li>
        </ul>
      </div>
      <div
        className="tabs-content"
        data-tabs-content="browse-by-tabs"
      >
        <div
          className="row margin-bottom-xxlarge"
        >
          <div
            className="small-4 columns combo text-center padding-top-medium"
          >
            <span>
              I am a card for 
              HCA
            </span>
             
            <br />
            <a
              href="link/hca"
            >
              <img
                className="margin-bottom-large experiment-thumbnail"
                height="800"
                src="test-file-stub"
              />
              <h6>
                <a
                  href="/gxa/experiments/E-MTAB-4344"
                >
                  Human tissues
                </a>
                <br />
                <a
                  href="/gxa/experiments/E-GEOD-26284"
                >
                  Human cells
                </a>
              </h6>
            </a>
          </div>
        </div>
      </div>
      <div
        className="row align-row-to-bottom"
      >
        <div
          className="small-6 small-centered columns margin-top-large"
        >
          <a
            className="button float-center"
            href="link/experiments"
            id="viewAllLink"
          >
            View all species
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
